### PR TITLE
Adds multiple-choice questions

### PR DESCRIPTION
**Pull request by jonnyspicer**
# Pull Request: Adds multiple-choice questions

## Changes Made
- Refactors large parts of `Predict.tsx` and related components, generally trying to achieve more granularity
- Adds new components related to MCQs
- Updates schema to handle MCQs
- Updates API and scoring logic to handle MCQs (calibration graph still missing currently)

## Testing
Tested locally and on preview build

### CODE REVIEW

File: styles/globals.css [hunk 1]
```diff
@@ -57,3 +57,19 @@ body,
   border-radius: 4px;
   background-color: rgba(0, 0, 0, 0.4);
 }
+
+/* I couldn't figure out how to do this in tailwind, maybe it's possible though */
+.child-flex-grow {
```
Comment (adam-binks) [inline styles/globals.css:62]: Looks like this is unused currently, could remove?

File: styles/globals.css [hunk 2]
```diff
@@ -57,3 +57,19 @@ body,
...
+  }
+}
+
+/* Similarly couldn't get the extended config working here */
+.grid-rows-1-3-1 {
+  grid-template-rows: 1fr 3fr 1fr;
+}
+
+.checkbox {
+  --chkbg: #6366f1;
```
[the remark this answers is not in the document]
Comment (jonnyspicer) [reply styles/globals.css:67]: This sets the background colour for `<input type="checkbox">` elements.

File: prisma/schema.prisma [hunk 3]
```diff
@@ -17,29 +23,47 @@ model Workspace {
...
 
 model QuestionScore {
-  id                  Int      @id @default(autoincrement())
-  createdAt           DateTime @default(now())
+  id                  Int             @id @default(autoincrement())
+  createdAt           DateTime        @default(now())
   relativeScore       Decimal?
   questionId          String
-  userQuestionComboId String   @unique
+  userQuestionComboId String          @unique
```
[the remark this answers is not in the document]
Comment (jonnyspicer) [reply prisma/schema.prisma:45]: Hmm I know why it works but your question points out that what I've done isn't ideal 😓 

At the moment the `userQuestionComboId` on a question is userid + questionid and only gets set if it's scored as a binary question. On options the questionid part of the combo is actually the optionid, and that only gets set if the option gets scored. This is pretty confusing, so I'll add a comment

[the remark this answers is not in the document]
Comment (adam-binks) [reply prisma/schema.prisma:45]: Sounds fine with a comment!

[the remark this answers is not in the document]
Comment (adam-binks) [reply prisma/schema.prisma:45]: (Sidenote - can you click "Resolve conversation" on this thread or do I need to?)

File: pages/api/calibration_graph.ts [hunk 4]
```diff
@@ -83,21 +111,45 @@ export async function getBucketedForecasts(userId: string, tags?: string[]) {
...
+          .filter(
+            (f) =>
+              !option.forecasts.some((f2) => {
+                const timeDiff = f2.createdAt.getTime() - f.createdAt.getTime()
+                return f !== f2 && timeDiff < 1000 * 60 && timeDiff > 0
+              }),
+          )
+          .map((f) => ({
+            forecast: f.forecast.toNumber(),
+            resolution: option.resolution,
```
[the remark this answers is not in the document]
Comment (jonnyspicer) [reply pages/api/calibration_graph.ts:147]: Great spot!

**Comment by adam-binks** [comment]
Merged (I was getting a bunch of incorrect conflicts, ended up just squashing)

**Pull request closed without merging**